build.sbt

```diff
@@ -1,17 +1,61 @@
-name := "morganey"
 
-organization := "me.rexim"
+/*
+ * =================================== settings ===================================
+ */
 
-version := "0.0.1-SNAPSHOT"
+lazy val commonSettings = Seq(
+  name := "morganey",
+  organization := "me.rexim",
+  version := "0.0.1-SNAPSHOT",
+  scalaVersion := "2.11.7",
+  scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature"),
+  libraryDependencies ++= Seq(
+    "jline" % "jline" % "2.6",
+    "org.scalatest" % "scalatest_2.11" % "2.2.4" % "test"
+  )
+)
 
-scalaVersion := "2.11.7"
+lazy val macroSettings = Seq(
+  name := "morganey-macros",
+  libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value
+)
 
-scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature")
+lazy val kernelSettings = Seq(
+  name := "morganey-kernel",
+  libraryDependencies += "org.scala-lang.modules" %% "scala-parser-combinators" % "1.0.4"
+)
 
```

Comment:
Macro Implementations have to be in a separate projects. They'll have to be compiled before the code, which uses the macro.
Because the macro uses your Parser and the definitions of AST-Nodes, the project structure has to be split. It might be possible to join the definitions of lambda-terms, parser and macros into one subproject, but macros and main-program (for example repl-implementation) have to be separate.